### PR TITLE
Sarkar/mediapipe sdxl

### PR DESCRIPTION
**Pull request by ssarkar2**
# What does this PR do?

1. Adds media pipe for SDXL (for 8x perf)
2. Adds flag to discount checkpoint saving time in throughput calculation `--discount_chkpoint_saving_in_throughput`

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/stable-diffusion/training/run_8x.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PT_HPU_RECIPE_CACHE_CONFIG=/tmp/stdxl_recipe_cache,True,1024  \
-python ../../gaudi_spawn.py --world_size 8 --use_mpi train_text_to_image_sdxl.py \
+python ../../gaudi_spawn.py --world_size 8 --use_mpi --master_port 29501 train_text_to_image_sdxl.py \
```
Comment (ssarkar2) [inline examples/stable-diffusion/training/run_8x.sh:2]: remove master_port

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 2]
```diff
@@ -515,6 +515,22 @@ def parse_args(input_args=None):
...
+        help="Use gaudi2 HW mediapipe over regular dataloader. \
+        case 1: nothing is passed to this argument -> regular torch dataloader is used\
+        case 2: an empty or non existant path is passed -> images are dumped from dataset (passed in through dataset_name) in that location before first run \
+        case 3: a non empty path is passed -> images from that location are used ",
+    )
+    parser.add_argument(
+        "--discount_chkpoint_saving_in_throughput",
+        default=False,
+        action="store_true",
+        help="Checkpoitn saving takes a lot of time. Ignore time for checkpoint saving for throughput calculations"
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:531]: can you align with https://github.com/huggingface/optimum-habana/blob/20832c92ce3e9cfe94aeb1797a5533b4be962a6c/optimum/habana/transformers/training_args.py#L187 ?

File: examples/stable-diffusion/training/train_text_to_image_sdxl.py [hunk 3]
```diff
@@ -515,6 +515,22 @@ def parse_args(input_args=None):
...
         help="Print the loss for every logging_step.",
     )
+    parser.add_argument(
+        "--mediapipe",
+        default="",
+        type=str,
+        help="Use gaudi2 HW mediapipe over regular dataloader. \
+        case 1: nothing is passed to this argument -> regular torch dataloader is used\
+        case 2: an empty or non existant path is passed -> images are dumped from dataset (passed in through dataset_name) in that location before first run \
+        case 3: a non empty path is passed -> images from that location are used ",
```
Comment (libinta) [inline examples/stable-diffusion/training/train_text_to_image_sdxl.py:525]: what will case 2 do?

**Review by ssarkar2 — COMMENTED** [review]
add license
change "discount" to "adjust_throughput"
add "imagesize" in requirements.txt

**Review by libinta — APPROVED** [review]

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/787

**Pull request merged**